src/cached_object_store.rs

```diff
@@ -94,8 +146,99 @@ impl CachedObjectStore {
         opts: GetOptions,
     ) -> object_store::Result<GetResult> {
         let get_range = opts.range.clone();
-        let (meta, attributes) = self.maybe_prefetch_range(location, opts).await?;
-        let range = self.canonicalize_range(get_range, meta.size)?;
+        let entry = self.cache_storage.entry(location, self.part_size_bytes);
+        match entry.read_head().await {
+            // Disk cache hit
+            Ok(Some((meta, attributes))) => {
+                let mut ghost_guard = self.ghost.lock().await;
+                self.decrease_ghost_capacity(&mut ghost_guard);
+                self.local_range_hit(location, get_range, meta, attributes)
+                    .await
+            }
+            // Disk cache miss, we try to read the object from the object store
+            Ok(None) => {
+                match self.maybe_prefetch_range(location, &opts).await {
+                    // Object is accessed for the first time, we will not store it in the disk cache
+                    Ok((Some(result), _, _)) => Ok(result),
+                    // Object is not accessed for the first time, we get result from disk cache
+                    Ok((None, meta, attributes)) => {
+                        self.local_range_hit(location, get_range, meta, attributes)
+                            .await
+                    }
+                    Err(e) => Err(e),
+                }
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    // TODO: implement the put with cache here
```

Comment:
is this implementation outside the scope of this PR?

Reply:
Oh, it is not relevant with this PR. I remember that @flaneur2020  added it.

Reply:
yeah, this todo is added by myself, this diff is a code move

it seems that some cache solution works not bad with these read-only cache, i suppose this todo is not in priority, and can be removed with replacing it into an github issue.